frontend/packages/dev-console/src/utils/pipeline-plugin-utils.ts

```diff
@@ -0,0 +1,70 @@
+import * as _ from 'lodash';
+import { referenceForModel, K8sResourceKind } from '@console/internal/module/k8s';
+import { FirehoseResource } from '@console/internal/components/utils';
+import { PipelineRunModel, PipelineModel } from '../models';
+import { Pipeline, PipelineRun } from './pipeline-augment';
+
+// label to get the pipelines
+export const PIPELINE_REF = 'app.openshift.io/pipeline-ref';
+
+export const tknPipelineAndPipelineRunsResources = (namespace: string): FirehoseResource[] => {
+  const resources = [
+    {
+      isList: true,
+      kind: referenceForModel(PipelineRunModel),
+      namespace,
+      prop: 'pipelineRuns',
+      optional: true,
+    },
+    {
+      isList: true,
+      kind: referenceForModel(PipelineModel),
+      namespace,
+      prop: 'pipelines',
+      optional: true,
+    },
+  ];
+  return resources;
+};
+
+type PipelineItem = {
+  pipeline: K8sResourceKind;
+  pipelineRuns: K8sResourceKind[];
+};
+
+const byCreationTime = (left, right) => {
```

Comment:
If we are keeping sort, you should type this function. `left` and `right` are not variable names that showcase type. Ideally when reading a function, there should be limited need to trace origins to understand the nature of the function.

Reply:
fixed in https://github.com/openshift/console/pull/3014/commits/65c3b41d3e36e8886063507b03df00ba95c78ab8